packages/ui-kit/scripts/parse-design-tokens.mts

```diff
@@ -0,0 +1,342 @@
+  const { type, value } = variable
+  switch (type) {
+    case 'color':
+      return `rgba(${value.r}, ${value.g}, ${value.b}, ${value.a})`
+    case 'number':
+      return `${value}px`
+    case 'effect': {
+      const { offset, radius, spread, color } = value.effects[0]
+      return `${offset.x}px ${offset.y}px ${radius}px ${spread}px rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`
+    }
+    default:
+      return value.toString()
+  }
+}
+
+// Parse Figma alias value
+const parseAlias = (variable: VariableProps, tokenData: TokenDataProps[]): string => {
+  const { value } = variable
+  const tokenKey = slugifyStr(value.name)
+  const tokenValue = tokenData.find((token) => token.cssName === `--propel-${tokenKey}`)
+  return tokenValue ? `var(${tokenValue.cssName})` : ''
+}
+
+// Generate design token data
+const generateTokenValue = (key: string, value: any, tokenData: TokenDataProps[]): TokenDataProps => ({
+  cssName: `--propel-${key}`,
+  jsName: kebabCaseToCamelCase(key),
+  kind: value.kind,
+  type: value.type === 'number' ? 'number' : 'string',
+  value: value.isAlias ? parseAlias(value, tokenData) : parseValue(value)
+})
+
+// Generate typography token data
+const generateTypographyValue = (key: string, type: 'number' | 'string', value: string): TokenDataProps =>
+  generateTokenValue(key, { type, value, kind: 'typography' }, [])
+
+// Write design token content to file
+const writeToFileSync = (fileName: string, content: string): void => {
+  const fullPath = path.join('./src/themes/generated/', fileName)
+  const dirName = path.dirname(fullPath)
+
+  // Check if the directory exists. If not, create it
+  if (!fs.existsSync(dirName)) {
+    fs.mkdirSync(dirName, { recursive: true })
+  }
+
+  fs.writeFileSync(fullPath, content)
+}
+
+// Read Figma variables from JSON file
+const getJSONFromFile = async (): Promise<VariablesJSONProps> => {
+  const data = await fsPromises.readFile('./src/themes/variables.json', 'utf-8')
+  const variablesJSON: VariablesJSONProps = JSON.parse(data)
+
+  return variablesJSON
+}
+
+// Get theme tokens
+
+const getThemeTokens = ({
+  name,
+  themes,
+  variables,
+  tokens
+}: {
+  name: string
+  themes?: ThemeProps[]
+  variables: TokenDataProps[]
+  tokens: TokenDataProps[]
+}) =>
+  themes
+    ?.find((theme) => theme.name.toLowerCase() === name)
+    ?.variables.map((variable) =>
+      generateTokenValue(
+        slugifyStr(variable.name.split('/').pop() ?? ''),
+        variable,
+        variable.isAlias ? variables : tokens
+      )
+    )
+
+const main = async () => {
+  const spinner = ora({ text: 'Parsing design variables and tokens...', color: 'yellow' }).start()
+
+  const variablesJSON = await getJSONFromFile()
+
+  if (!variablesJSON) {
+    spinner.fail('Failed to parse variables.json')
+    throw new Error('Failed to parse variables.json')
+  }
+
+  try {
+    const variables: TokenDataProps[] = []
+    const tokens: TokenDataProps[] = []
```

Comment:
Maybe we can improve error handling here and set this inside the try catch as well, `getJSONFromFile` could fail from `JSON.parse` and `fsPromises.readFile`

```suggestion
  try {
    const variablesJSON = await getJSONFromFile()

    if (!variablesJSON) {
      throw new Error('Failed to get variables.json file')
    }

    const variables: TokenDataProps[] = []
    const tokens: TokenDataProps[] = []
```